ykrt/src/compile/jitc_yk/codegen/x86_64/mod.rs

```diff
@@ -249,10 +249,14 @@ impl<'a> Assemble<'a> {
             for (deoptid, fail_label) in fail_labels.into_iter().enumerate() {
                 #[cfg(any(debug_assertions, test))]
                 self.comment(self.asm.offset(), format!("Deopt ID for guard {deoptid}"));
+                // FIXME: Why are `deoptid`s 64 bit? We're not going to have that many guards!
```

Comment:
Because we just pass them through from LLVM's stackmap where they are `i64`.

Reply:
Actually, that's not true. They are not the same as stackmap IDs. So we can turn them into `i32`s if you like.

Reply:
I *think* these are internal to yk, so they're per-trace? If so, we could probably get away with `u16`s! I guess we leave things as-is for this PR and fix this later?